core/src/main/scala/chisel3/internal/Binding.scala

```diff
@@ -103,42 +112,51 @@ sealed trait ConditionalDeclarable extends TopBinding {
 
 // TODO(twigg): Ops between unenclosed nodes can also be unenclosed
 // However, Chisel currently binds all op results to a module
-
+@deprecated(deprecatedPublicAPIMsg, "Chisel 3.6")
 case class PortBinding(enclosure: BaseModule) extends ConstrainedBinding
 
+@deprecated(deprecatedPublicAPIMsg, "Chisel 3.6")
 case class OpBinding(enclosure: RawModule, visibility: Option[WhenContext])
     extends ConstrainedBinding
     with ReadOnlyBinding
     with ConditionalDeclarable
+@deprecated(deprecatedPublicAPIMsg, "Chisel 3.6")
 case class MemoryPortBinding(enclosure: RawModule, visibility: Option[WhenContext])
     extends ConstrainedBinding
     with ConditionalDeclarable
+@deprecated(deprecatedPublicAPIMsg, "Chisel 3.6")
 case class RegBinding(enclosure: RawModule, visibility: Option[WhenContext])
     extends ConstrainedBinding
     with ConditionalDeclarable
+@deprecated(deprecatedPublicAPIMsg, "Chisel 3.6")
 case class WireBinding(enclosure: RawModule, visibility: Option[WhenContext])
     extends ConstrainedBinding
     with ConditionalDeclarable
 
+@deprecated(deprecatedPublicAPIMsg, "Chisel 3.6")
 case class ChildBinding(parent: Data) extends Binding {
   def location: Option[BaseModule] = parent.topBinding.location
 }
 
 /** Special binding for Vec.sample_element */
+@deprecated(deprecatedPublicAPIMsg, "Chisel 3.6")
 case class SampleElementBinding[T <: Data](parent: Vec[T]) extends Binding {
   def location = parent.topBinding.location
 }
 
 /** Special binding for Mem types */
+@deprecated(deprecatedPublicAPIMsg, "Chisel 3.6")
 case class MemTypeBinding[T <: Data](parent: MemBase[T]) extends Binding {
   def location: Option[BaseModule] = parent._parent
 }
 // A DontCare element has a specific Binding, somewhat like a literal.
 // It is a source (RHS). It may only be connected/applied to sinks.
+@deprecated(deprecatedPublicAPIMsg, "Chisel 3.6")
 case class DontCareBinding() extends UnconstrainedBinding
 
 // Views currently only support 1:1 Element-level mappings
-private[chisel3] case class ViewBinding(target: Element) extends UnconstrainedBinding
+@deprecated(deprecatedPublicAPIMsg, "Chisel 3.6")
+case class ViewBinding(target: Element) extends UnconstrainedBinding
```

Comment:
```suggestion
private[chisel3] case class ViewBinding(target: Element) extends UnconstrainedBinding
```
This one is already private, why are you making it public?